homeassistant/components/mqtt/__init__.py

```diff
@@ -716,7 +716,8 @@ def init_client(self):
             self._mqttc.will_set(  # pylint: disable=no-value-for-parameter
                 *attr.astuple(
                     will_message,
-                    filter=lambda attr, value: attr.name != "subscribed_topic",
+                    filter=lambda attr, value: attr.name
+                    not in ["subscribed_topic", "timestamp"],
```

Comment:
These are a little dangerous and will break again in the future. We might want to consider in a future PR to just type `topic=will_message.bla, payload=will_message.payload` etc.

Reply:
Yeah, you're right. OK to merge this for 0.113.0 though?